cmd/kubelet/app/server.go

```diff
@@ -845,6 +845,17 @@ func buildKubeletClientConfig(ctx context.Context, s *options.KubeletServer, nod
 		if err != nil {
 			return nil, nil, err
 		}
+		// Kubelet needs to be able to recover from stale http connections.
+		// HTTP2 has a mechanism to detect broken connections by sending periodical pings.
+		// HTTP1 only can have one persistent connection, and it will close all Idle connections
+		// once the Kubelet heartbeat fails. However, since there are many edge cases that we can't
+		// control, users can still opt-in to the previous behavior for closing the connections by
+		// setting the environment variable DISABLE_HTTP2.
+		if s := os.Getenv("DISABLE_HTTP2"); len(s) > 0 {
+			klog.InfoS("HTTP2 has been explicitly disabled, Kubelet will forcefully close active connections on heartbeat failures")
+		} else {
+			closeAllConns = func() { utilnet.CloseIdleConnectionsFor(transportConfig.Transport) }
+		}
```

Comment:
since the function we're returning might not actually be closing all connections now, let's name the return var what it actually is (same below with the existing code, and same with the var the return value from this function is assigned to in `run()`)

```diff
+		var onHeartbeatFailure func()
		if s := os.Getenv("DISABLE_HTTP2"); len(s) > 0 {
			klog.InfoS("HTTP2 has been explicitly disabled, Kubelet will forcefully close active connections on heartbeat failures")
+			onHeartbeatFailure = closeAllConns
		} else {
-			closeAllConns = func() { utilnet.CloseIdleConnectionsFor(transportConfig.Transport) }
+			onHeartbeatFailure = func() { utilnet.CloseIdleConnectionsFor(transportConfig.Transport) }
		}
...
-		return transportConfig, closeAllConns, nil
+ 		return transportConfig, onHeartbeatFailure, nil
```

Reply:
added a new commit with the renames so all codebase is consistent with the name